### PR TITLE
Fix for 255639 and mtree fixes

### PR DESCRIPTION
**Pull request by jlduran**
The main idea is to fix [255639](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=255639), and clean up a bit while doing it.

cc/ @bsdimp @kevans91 @emaste

### CODE REVIEW

File: etc/mtree/README [hunk 1]
```diff
@@ -23,8 +23,8 @@ and keep them in good working order.
 
             mkdir /tmp/MTREE
             mtree -deU -f BSD.X.dist -p /tmp/MTREE
-            mtree -cdin -k uname,gname,mode -p /tmp/MTREE | \
-		sed -e 's/ *$//' >BSD.X.new
+            mtree -bcdijn -k uname,gname,mode -p /tmp/MTREE | \
```
Comment (jlduran) [inline etc/mtree/README:26]: These instructions were probably designed for fmtree. If so, I'll update the flags accordingly.
`-i` -> `-j`

**Comment by bsdimp** [comment]
These changes look good to my eye. I'm not quite sure how to make this a 'approved review comment' though...

**Comment by bsdimp** [comment]
Started a full buildworld / installworld (ditto kernel) before committing this change....  should be done soonish

**Comment by bsdimp** [comment]
landed. this tested right for me, but be on the lookout on IRC/discord/social media + email for any weird reports...  this stuff seemed low risk and {build,install}{world,kernel} just worked, which is the biggest risk from this change.

**Pull request closed without merging**